Assets/Scripts/Navigation/NavigationManager.cs

```diff
@@ -0,0 +1,45 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class NavigationManager : MonoBehaviour
+{
+    public static NavigationManager Instance;
+
+    // save the data in a list
+    public List<string> sceneQueue = new List<string>();
+
+    private void Awake()
+    {
+        if (Instance != null)
+        {
+            Destroy(gameObject);
+            return;
+        }
+
+        Instance = this;
+        DontDestroyOnLoad(gameObject);
+    }
+
+    public void LoadNewScene(string sceneName)
+    {      
+        // grabs the current scene and appends it to the list queue
+        string currScene = SceneManager.GetActiveScene().name;
+        sceneQueue.Add(currScene);
+        
+        // goes to next scene
+        SceneManager.LoadScene(sceneName);
+    }
+
+    public void returnToPrevScene() 
+    {
+        // retrieves and pops the last scene from the list
+        string prevScene = sceneQueue[sceneQueue.Count - 1];
```

Comment:
Instead of sceneQueue[sceneQueue.Count-1] I believe you can just call sceneQueue.Last(). The benefit is you don't risk index out of range exceptions